internal/pkg/manifest/pkger/pkg.go

```diff
@@ -0,0 +1,91 @@
+// Package pkger provides functionality to transform an application manifest
+// into structures that can be consumed by infrastructure-as-code providers.
+package pkger
+
+import (
+	"fmt"
+
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/archer"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/aws/cloudformation"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/aws/session"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/deploy"
+	deploycfn "github.com/aws/amazon-ecs-cli-v2/internal/pkg/deploy/cloudformation"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/deploy/cloudformation/stack"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/manifest"
+)
+
+type projectDecriber interface {
+	GetProjectResourcesByRegion(project *archer.Project, region string) (*archer.ProjectRegionalResources, error)
+}
+
+type WebAppPkger struct {
+	app      *manifest.LBFargateManifest
+	env      *archer.Environment
+	project  *archer.Project
+	imageTag string
+
+	projectDecriber projectDecriber
+}
+
+func NewWebAppPkger(app *manifest.LBFargateManifest, env *archer.Environment, project *archer.Project, imageTag string) (*WebAppPkger, error) {
```

Comment:
Another option is to have an app_packager which takes in an AppManifest. This then uses the name + type casting to choose the correct, specific packager. 

One thing I'd really really like from these packagers is if you can just add one file for a new pattern - I'd optimize for ease of adding. 

So maybe AppPkger could have a map[string]*CloudFormationPkger or whatever it looks like. Then in each packager file, it could do a Registry[manifest.LoadBalancedWebApplication] = NewWebAppPkger

Just thinking out loud. That way every time we add a new App Type - we add the manifest and then we add the pklger, but minimize the number of files we have to edit.